include/swift/Reflection/ReflectionContext.h

```diff
@@ -474,8 +494,30 @@ class ReflectionContext
     savedBuffers.push_back(std::move(Buf));
     return true;
   }
-         
-  bool readELF(RemoteAddress ImageStart) {
+
+  /// Parses metadata information from an ELF image. Because the Section
+  /// Header Table is stripped out from the running process, this method also
+  /// receives a buffer with the contents of the image's file, from
+  /// where it will read the missing information.
+  ///
+  ///
+  /// \param[in] ImageStart
+  ///     A remote address pointing to the start of the image in the running
+  ///     process.
+  ///
+  /// \param[in] FileBuffer
+  ///     A buffer which contains the contents of the image's file
+  ///     in disk. The function returns false if the buffer is null.
+  ///
+  /// \param[in] FileSize
+  ///     The size of \p FileBuffer. The function returns false if the size is
+  ///     0.
+  ///
+  /// \return
+  ///     /b True if the metadata information was parsed successfully,
+  ///     /b false otherwise.
+  bool readELF(RemoteAddress ImageStart, const void *FileBuffer,
+               uint64_t FileSize) {
```

Comment:
Oh! I just realized that this header file has access to LLVM. In that case FileBuffer should probably be an llvm::StringRef (or a llvm::sys::MemoryBlock, which I just discovered).